src/utils/thread_helper.cpp

```diff
@@ -944,6 +944,44 @@ void ThreadHelper::DetachGracefullyCallback(void)
 
 #if OTBR_ENABLE_TELEMETRY_DATA_API
 #if OTBR_ENABLE_BORDER_ROUTING
+void ThreadHelper::RetrieveInfraLinkInfo(threadnetwork::TelemetryData::InfraLinkInfo *aInfraLinkInfo)
+{
+    {
+        otSysInfraNetIfAddressCounters addressCounters;
+        uint32_t                       ifrFlags = otSysGetInfraNetifFlags();
+
+        otSysCountInfraNetifAddresses(&addressCounters);
+
+        aInfraLinkInfo->set_name(otSysGetInfraNetifName());
+        aInfraLinkInfo->set_is_up((ifrFlags & IFF_UP) != 0);
+        aInfraLinkInfo->set_is_running((ifrFlags & IFF_RUNNING) != 0);
+        aInfraLinkInfo->set_is_multicast((ifrFlags & IFF_MULTICAST) != 0);
+        aInfraLinkInfo->set_link_local_address_count(addressCounters.mLinkLocalAddresses);
+        aInfraLinkInfo->set_unique_local_address_count(addressCounters.mUniqueLocalAddresses);
+        aInfraLinkInfo->set_global_unicast_address_count(addressCounters.mGlobalUnicastAddresses);
+    }
+
+    //---- infra_br_count
+    {
+        uint8_t                            count = 0;
+        otBorderRoutingPrefixTableIterator iterator;
+        otBorderRoutingRouterEntry         entry;
+
+        otBorderRoutingPrefixTableInitIterator(mInstance, &iterator);
+
+        while (otBorderRoutingGetNextRouterEntry(mInstance, &iterator, &entry) == OT_ERROR_NONE)
```

Comment:
I don't think `otBorderRoutingGetNextRouterEntry` only returns BRs. It also returns ordinary WiFi/Ethernet routers.

Reply:
Thanks for pointing this out! I will update a new version

Reply:
updated